src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -35,104 +32,28 @@ def is_local_kind(symbol_type: ts.FieldType) -> bool:
-    +------------------------------------+------------------------+
-    """
-    assert not isinstance(symbol_type, ts.DeferredType)
-    if isinstance(symbol_type, ts.FieldType):
-        return ITIRTypeKind.ITERATOR
-    elif any(type_info.primitive_constituents(symbol_type).if_isinstance(ts.FieldType)):
-        # if we encounter any field type that is defined on a local dimension
-        #  the resulting type on iterator ir level is not an iterator, but contains
-        #  one, e.g. a tuple of iterators.
-        if any(
-            type_info.primitive_constituents(symbol_type)
-            .if_isinstance(ts.FieldType)
-            .filter(is_local_kind)
-        ):
-            return ITIRTypeKind.ENCAPSULATED_ITERATOR
-        # otherwise we get an iterator, e.g. an iterator of values or tuples
-        return ITIRTypeKind.ITERATOR
-    return ITIRTypeKind.VALUE
-
-
-def is_expr_with_iterator_type_kind(it_type_kind: ITIRTypeKind) -> Callable[[foast.Expr], bool]:
-    def predicate(node: foast.Expr):
-        return iterator_type_kind(node.type) is it_type_kind
-
-    return predicate
-
-
-def to_value(node: foast.Expr) -> Callable[[itir.Expr], itir.Expr]:
-    """
-    Either ``deref_`` or noop callable depending on the input node.
-
-    Input node must have a scalar, non-local field, or tuple of non-local fields
-    type. If the lowered input node will represent an iterator expression,
-    return ``deref_``. Otherwise return a noop callable.
-
-    Examples:
-    ---------
-    >>> from gt4py.next.ffront.func_to_foast import FieldOperatorParser
-    >>> from gt4py.next.ffront.fbuiltins import float64
-    >>> from gt4py.next.common import Field, Dimension
-    >>> IDim = Dimension("IDim")
-    >>> def foo(a: Field[[IDim], "float64"]):
-    ...    b = 5
-    ...    return a, b
-
-    >>> parsed = FieldOperatorParser.apply_to_function(foo)
-    >>> field_a, scalar_b = parsed.body.stmts[-1].value.elts
-    >>> to_value(field_a)(im.ref("a"))
-    FunCall(fun=SymRef(id=SymbolRef('deref')), args=[SymRef(id=SymbolRef('a'))])
-    >>> to_value(scalar_b)(im.ref("a"))
-    SymRef(id=SymbolRef('a'))
-    """
-    if iterator_type_kind(node.type) is ITIRTypeKind.ITERATOR:
-        # just to ensure we don't accidentally deref a local field
-        assert not (isinstance(node.type, ts.FieldType) and is_local_kind(node.type))
-        return im.deref_
-    elif iterator_type_kind(node.type) is ITIRTypeKind.VALUE:
-        return lambda x: x
-
-    raise AssertionError(f"Type {node.type} can not be turned into a value.")
+def promote_to_list(
+    node: foast.Symbol | foast.Expr,
+) -> Callable[[itir.Expr], itir.Expr]:
+    if not is_local_type_kind(node.type):
+        return lambda x: im.as_lifted_lambda("make_const_list", x)
+    return lambda x: x
 
 
 class FieldOperatorLowering(NodeTranslator):
     """
     Lower FieldOperator AST (FOAST) to Iterator IR (ITIR).
 
+    The strategy is to lower every expression to lifted stencils,
```

Comment:
This change in approach invalidates `docs/development/ADRs/0002-Field_View_Lowering.md` (already outdated before this PR). Might be something I should take care of. Let's discuss in person.

Reply:
I marked the ADR as deprecated.